### ISSUE TITLE
an error when loading the module in webpack

### ISSUE THREAD

**Issue by jhlosin** [vote539/socketio-file-upload#41]
It gives "define cannot be used indirect" error when trying to import it in webpack environment.
Is this a compatibility issue? any helpful comments I can get?

webpack@1.12.9

**Comment by vote539** [comment]
I've never attempted using this module with webpack.  If you figure out what the problem is, submit a PR. :)

**Comment by MaffooBristol** [comment]
This seem to be a common issue [here](https://github.com/webpack/webpack/issues/138) and [here](https://github.com/request/request/issues/1920) and elsewhere too. Seems the issue may with the module-loading closure code that SIOFU uses, although it's quite a common pattern, perhaps not any more if it breaks webpack and stuff.

What version of webpack are you running? Are you trying to do `require()` or `import from`?

All the issues and tickets I've seen so far seem to have the hacky workaround of just not parsing certain modules, but that doesn't seem ideal in the long run.

**Comment by vote539** [comment]
Thanks @MaffooBristol for doing more research on this!

The module loading pattern used here is the same one that is used by dozens of other modules, as mentioned in the tickets you linked.  I'm not an expert on WebPack so I don't know what the right solution would be.

For what it's worth, in my workflow I use [require.js optimizer](http://requirejs.org/docs/optimization.html) which works without issues on SIOFU.

**Comment by Sjiep** [comment]
I too am running into this problem. I'll see what I can do to find a fix!

**Comment by Sjiep** [comment]
As @MaffooBristol mentioned, webpack cannot load SIOFU because of the module-loading closure. 
I commented out this part in the code.
```
(function (scope, name, factory) {
	if (typeof define === "function" && define.amd) {
		define(name, factory);
	}
	else if (typeof module === 'object' && module.exports) {
		module.exports = factory();
	}
	else {
		scope[name] = factory();
	}
}(this, "SocketIOFileUpload", function () {
```
and replaced it with: 

```
SocketIOFileUpload = function (socket) {
   ....
}
module.exports = SocketIOFileUpload
```

This made it work for webpack. However by doing this does it removes the AMD support. I would have to check how we can support both...

**Comment by vote539** [comment]
Fixed by #55.
